### PR TITLE
DDBTEAM-389: Added abstrct tsv vendor

### PR DESCRIPTION
**Pull request by cableman**

### CODE REVIEW

**Comment by cableman** [comment]
The `phpcsfixerr` is in fact not an error.

File: src/Service/VendorService/AbstractTsvVendorService.php [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+{
+    use ProgressBarTrait;
+
+    protected $vendorArchiveDir = 'AbstractTsvVendor';
+    protected $vendorArchiveName = 'covers.tsv';
+
+    private $resourcesDir;
+
+    /**
+     * SaxoVendorService constructor.
```
Comment (tuj) [inline src/Service/VendorService/AbstractTsvVendorService.php]: Saxo?

File: src/Service/VendorService/AbstractTsvVendorService.php [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+            $totalRows = 0;
+            $pidArray = [];
+
+            foreach ($reader->getSheetIterator() as $sheet) {
+                foreach ($sheet->getRowIterator() as $row) {
+                    $cellsArray = $row->getCells();
+                    $fields = $this->findCellName($cellsArray);
+                    if (0 === $totalRows) {
+                        // Fist row in the tsv file contains the headers.
```
Comment (tuj) [inline src/Service/VendorService/AbstractTsvVendorService.php]: Fist?

File: src/Service/VendorService/AbstractTsvVendorService.php [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+                        }
+                    }
+
+                    ++$totalRows;
+
+                    if ($this->limit && $totalRows >= $this->limit) {
+                        break;
+                    }
+
+                    if (0 === $totalRows % 100) {
```
Comment (tuj) [inline src/Service/VendorService/AbstractTsvVendorService.php]: 100?

File: src/Service/VendorService/AbstractTsvVendorService.php [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+            $reader = $this->getSheetReader();
+
+            $totalRows = 0;
+            $pidArray = [];
+
+            foreach ($reader->getSheetIterator() as $sheet) {
+                foreach ($sheet->getRowIterator() as $row) {
+                    $cellsArray = $row->getCells();
+                    $fields = $this->findCellName($cellsArray);
```
Comment (tuj) [inline src/Service/VendorService/AbstractTsvVendorService.php]: Move inside if

File: src/Service/VendorService/AbstractTsvVendorService.php [hunk 5]
```diff
@@ -0,0 +1,157 @@
+<?php
+/**
+ * @file
+ * Abstract vendor for tsv file imports.
+ */
+
+namespace App\Service\VendorService;
+
+use App\Exception\UnknownVendorResourceFormatException;
+use App\Service\VendorService\AbstractBaseVendorService;
```
Comment (tuj) [inline src/Service/VendorService/AbstractTsvVendorService.php]: Remove

File: src/Service/VendorService/AbstractTsvVendorService.php (hunk 1 above)
Comment (cableman) [reply src/Service/VendorService/AbstractTsvVendorService.php]: Fixed

File: src/Service/VendorService/AbstractTsvVendorService.php (hunk 2 above)
Comment (cableman) [reply src/Service/VendorService/AbstractTsvVendorService.php]: Fixed

File: src/Service/VendorService/AbstractTsvVendorService.php (hunk 3 above)
Comment (cableman) [reply src/Service/VendorService/AbstractTsvVendorService.php]: Fixed

File: src/Service/VendorService/AbstractTsvVendorService.php (hunk 4 above)
Comment (cableman) [reply src/Service/VendorService/AbstractTsvVendorService.php]: Fixed

File: src/Service/VendorService/AbstractTsvVendorService.php (hunk 5 above)
Comment (cableman) [reply src/Service/VendorService/AbstractTsvVendorService.php]: Fixed

**Pull request merged**